extensions/wikia/SpitfiresContributionExperiments/scripts/my-profile-tour.js

```diff
@@ -0,0 +1,177 @@
+			userName = mw.user.name(),
+			userPage = 'User:' + userName.replace(/ /g, '_'),
+			templateData = {
+				userName: userName
+			},
+			lastStepTemplateData = [];
+
+		function init() {
+			if ($.cookie(seenCookieName)) {
+				return;
+			}
+			$.when(
+				nirvana.sendRequest({
+					controller: 'ContributionExperiments',
+					method: 'getNextPages',
+					type: 'get'
+				}),
+				loader({
+					type: loader.MULTI,
+					resources: {
+						mustache: '/extensions/wikia/SpitfiresContributionExperiments/templates/MyProfileTourExperimentModal.mustache,' +
+						'/extensions/wikia/SpitfiresContributionExperiments/templates/MyProfileTourExperimentStep1.mustache,' +
+						'/extensions/wikia/SpitfiresContributionExperiments/templates/MyProfileTourExperimentStep2.mustache,' +
+						'/extensions/wikia/SpitfiresContributionExperiments/templates/MyProfileTourExperimentStep3.mustache,' +
+						'/extensions/wikia/SpitfiresContributionExperiments/templates/MyProfileTourExperimentStep4.mustache',
+						styles: '/extensions/wikia/SpitfiresContributionExperiments/styles/my-profile-tour.scss'
+					}
+				})
+			).done(renderModal);
+		}
+
+		function renderModal(data, resources) {
+			$.cookie(seenCookieName, 1, {expires: 30});
+			templates = resources.mustache;
+			lastStepTemplateData = data[0];
+
+			loader.processStyle(resources.styles);
+			modalConfig.vars.content = mustache.render(resources.mustache[0], {});
+
+			require(['wikia.ui.factory'], function (uiFactory) {
+				uiFactory.init(['modal']).then(createModalComponent);
+			});
+		}
+
+		function createModalComponent(uiModal) {
+			uiModal.createComponent(modalConfig, processInstance);
+		}
+
+		function processInstance(modalInstance) {
+			var modalContent = modalInstance.$element.find('.my-profile-content'),
+				$profileModal;
+
+			modal = modalInstance;
+
+			modalInstance.show();
+
+			modalContent.html(mustache.render(templates[currentStep], templateData));
+
+			tracker.trackVerboseImpression(experimentName, 'step-1-first-time');
+
+			$profileModal = $('#MyProfileModal');
+			$profileModal.on('click', '.next-step', function () {
+				addAnswer();
+
+				currentStep++;
+				templateData.answer = answers[currentStep];
+				if (currentStep === 4) {
+					sendProfileData().done(function () {
+						modalContent.html(mustache.render(templates[currentStep], lastStepTemplateData));
+					});
+				} else {
+					modalContent.html(mustache.render(templates[currentStep], templateData));
+				}
+
+				tracker.trackVerboseClick(experimentName, 'go-next-to-step-' + currentStep);
+			});
+
+			$profileModal.on('click', '.my-profile-header-go-back', function () {
+				addAnswer();
+				currentStep--;
+				templateData.answer = answers[currentStep];
+				modalContent.html(mustache.render(templates[currentStep], templateData));
+
+				tracker.trackVerboseClick(experimentName, 'go-back-to-step-' + currentStep);
+			});
+
+			$profileModal.on('click', '.my-profile-next-pages a', function () {
```

Comment:
Since this involves navigating away, the preferred event is `.on('mousedown touchstart', ...` to allow the browser a little more time to send the tracker calls through.